recirq/quantum_chess/circuit_transformer.py

```diff
@@ -181,27 +193,32 @@ def map_helper(self,
         # This makes back-tracking easier.
         node_to_map = nodes_to_map[0]
 
+
         for node_to_try in valid_adjacent_qubits:
             # Add proposed qubit to the mapping
             # and remove it from available qubits
             mapping[node_to_map] = node_to_try
             available_qubits.remove(node_to_try)
-
+            nodes_trying.append(node_to_map)
             # Recurse
             # Move on to the qubit we just mapped.
             # Then, come back to this node and
             # map the rest of the adjacent nodes
 
             success = self.map_helper(node_to_map, mapping, available_qubits,
-                                      graph)
+                                      graph, nodes_trying)
             if success:
                 success = self.map_helper(cur_node, mapping, available_qubits,
-                                          graph)
+                                          graph, nodes_trying)
 
             if not success:
                 # We have failed.  Undo this mapping and try another one.
-                del mapping[node_to_map]
-                available_qubits.add(node_to_try)
+                while True:
```

Comment:
Should this check if nodes_trying is not empty instead of just being True?

Reply:
named_qubit == node_to_map will always be satisfied by one named_qubit in nodes_trying, so I think the current condition is fine.